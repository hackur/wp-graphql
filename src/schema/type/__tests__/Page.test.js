import PageType from 'type/Page';

// eslint-disable-next-line no-underscore-dangle
const fields = PageType._typeConfig.fields();

describe('Test Page type', () => {
  test('Test name', () => {
    expect(PageType.name).toMatchSnapshot();
  });

  test('Test type', () => {
    expect(PageType.isTypeOf({ type: 'page' })).toBe(true);
  });

  test('Test fields', () => {
    expect(Object.keys(PageType.getFields())).toMatchSnapshot();
  });

  test('Test resolve parent', async () => {
    const parent = await fields.parent.resolve({ parent: 69 });
    expect(parent).toMatchSnapshot();
  });

  test('Test resolve parent null', () => {
    expect(fields.parent.resolve({ parent: 0 })).toBeNull();
  });
});
