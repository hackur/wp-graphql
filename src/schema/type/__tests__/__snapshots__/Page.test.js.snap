// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Page type Test fields 1`] = `
Array [
  "id",
  "date",
  "date_gmt",
  "guid",
  "modified",
  "modified_gmt",
  "slug",
  "type",
  "link",
  "title",
  "content",
  "excerpt",
  "comment_status",
  "ping_status",
  "template",
  "author",
  "featuredMedia",
  "meta",
  "parent",
  "menu_order",
]
`;

exports[`Test Page type Test name 1`] = `"Page"`;

exports[`Test Page type Test resolve parent 1`] = `
Page {
  "id": 69,
}
`;
